### DISCUSSION TITLE
Customizing Shortcuts Definition

### DISCUSSION THREAD

**Discussion by milisarge** [CuarzoSoftware/Louvre#63] [Support] [answered]
When i look at the F1 key, it is predefined with weston-terminal (https://github.com/CuarzoSoftware/Louvre/blob/980bf1a06db95b94b22e46e07ffb25f15380129e/src/lib/core/default/LKeyboardDefault.cpp#L47), 

If i want to re-define it with my terminal application, which way is the proper way of doing it? 

By the way, i am using Louvre core library, just curious why keyboard definitions are embed in the core library?

**Comment by milisarge** [comment]
I forked my project from loure-views, i defined the key in Keyboard.cpp but it also triggers core definition, my terminal app and weston-terminal are opening at the same time when i press F1 :) i would like to modify without hacking the core lib.

**Comment by Fox2Code** [comment]
Louvre View should have disabled F1 key from this from this:

https://github.com/CuarzoSoftware/Louvre/blob/main/src/examples/louvre-views/src/Keyboard.cpp
https://github.com/CuarzoSoftware/Louvre/blob/main/src/examples/louvre-views/src/Compositor.cpp#L142

This is weird, definitely needs more investigating.

**Comment by ehopperdietzel** [comment, marked as answer]
There are custom [flags](https://cuarzosoftware.github.io/Louvre/class_louvre_1_1_l_scene.html#a10f11d8257c27295b313726fe40188f5) you can pass to LScene event handlers. Check [LScene::handleKeyboardKeyEvent()](https://cuarzosoftware.github.io/Louvre/class_louvre_1_1_l_scene.html#af8678b253782a6355c0e43138a563575), so you don't have to write it yourself.

By default, all are set. In this case, you should remove the **AuxFunc** flag.

```cpp
scene.handleKeyboardKeyEvent(event, LScene::WaylandEvents);
```

Keep in mind that other shortcuts, such as exiting the compositor with `Ctrl + Shift + Esc`, will also be disabled.

Here is the default implementation: [LKeyboard::keyEvent()](https://cuarzosoftware.github.io/Louvre/class_louvre_1_1_l_keyboard.html#a6be23845ffc211d3fe8ea84f7b90bcbd)

**Comment by Fox2Code** [comment]
I think this detail should be added to the base tutorial/documentation.

**Comment by ehopperdietzel** [comment]
It is https://github.com/CuarzoSoftware/LouvreTemplate/blob/main/src/utils/Settings.h

**Comment by milisarge** [comment]
Sorry, i could not follow how can i overwriting the F1 key binding by using `LScene::handleKeyboardKeyEvent`

**Comment by ehopperdietzel** [comment]
You need to explicitly set `LScene::WaylandEvents` as the second argument of `LScene::handleKeyboardKeyEvent()`.
